app/models/user.rb

```diff
@@ -0,0 +1,21 @@
+class User < ActiveRecord::Base
+  has_many :tweets
+  has_secure_password
+
+  def slug
+    username.downcase.gsub(" ","-")
+  end
+
+  def self.find_by_slug(slug)
+    User.all.find{|user| user.slug == slug}
+  end
+
+  def self.current_user(session)
```

Comment:
Implementing `current_user` and `is_logged_in` as controller helper methods would make them easier to invoke and would allow you to avoid passing the `session` object as a parameter.